src/app/_components/NewsletterForm.tsx

```diff
@@ -0,0 +1,64 @@
+'use client'
+
+import React from 'react'
+
+import { zodResolver } from '@hookform/resolvers/zod'
+import { FieldErrors, SubmitHandler, useForm } from 'react-hook-form'
+
+import { NewsletterSchema } from '@/_schemas/newsletterSchema'
+
+import { Button } from './Button'
+import ControlledInput from './ControlledInput'
+import Form from './Form'
+import { InputProps } from './Input'
+
+export type FormType = {
+  email: string
+}
+
```

Comment:
it looks like we can infer the FormType from the schema, so that we avoid duplicating information.

What do you think?

```suggestion
export type FormType = z.infer<typeof NewsletterSchema>
```

Reply:
To make this work properly, we have to remove `ZodSchema` from `export const NewsletterSchema`.

Otherwise, TypeScript cannot infer the type and gives us `FormType = any`.